### PR TITLE
Added documentation for FASTQ datatypes and implemented quality check

### PR DESCRIPTION
**Pull request by bernt-matthias**
## What did you do? 

- Added documentation on the definition of the FASTQ data types
- Implemented quality check for the (sequence space) FASTQ data types

Definitions are based on 

- https://github.com/galaxyproject/sequence_utils/blob/master/galaxy_utils/sequence/fastq.py (ignoring the maxima)
- https://en.wikipedia.org/wiki/FASTQ_format#Encoding (keeping the current maximum quality of 50 for fastqsanger)

- [ ] Still confused by https://galaxyproject.org/learn/datatypes/#Fastq that uses integers for the qualities in the solexa example. 
- [ ] Maybe we should update the link and use it always as `description_url`

## Why did you make this change?

- Definition may be helpful for tool developers. 
- Stricter check for the format may forbid some wrong assignments by users. 

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [ ] I've included a screenshot of the changes

### CODE REVIEW

File: lib/galaxy/datatypes/sequence.py [hunk 1]
```diff
@@ -844,19 +847,45 @@ def quality_check(lines):
...
+    solexa scored quality values -5:40 represented by ASCII 59:104
+    """
     edam_format = "format_1933"
     file_ext = "fastqsolexa"
 
+    @staticmethod
+    def quality_check(lines):
+        """Presuming lines are lines from a fastq file, return True if the qualities are compatible with sanger encoding"""
+        for line in islice(lines, 3, None, 4):
+            if not all(_ >= ';' and _ <= 'h' for _ in line[0]):
```
Comment (jdavcs) [inline lib/galaxy/datatypes/sequence.py]: I think, the underscore, by convention, is designated for unused variables (placeholders in unpacking, listcomps, etc.). (PEP 640 is proposing to formalize this). But here the `_` variable is used for comparison and, therefore, has meaning. My suggestion would be to give it a real/descriptive name (a single-character name would be better, I think).

File: lib/galaxy/datatypes/sequence.py [hunk 2]
```diff
@@ -844,19 +847,45 @@ def quality_check(lines):
...
+    phred scored quality values 0:40 represented by ASCII 64:104
+    """
     edam_format = "format_1931"
     file_ext = "fastqillumina"
 
+    @staticmethod
+    def quality_check(lines):
+        """Presuming lines are lines from a fastq file, return True if the qualities are compatible with sanger encoding"""
+        for line in islice(lines, 3, None, 4):
+            if not all(_ >= '@' and _ <= 'h' for _ in line[0]):
```
Comment (jdavcs) [inline lib/galaxy/datatypes/sequence.py]: same as above

File: lib/galaxy/datatypes/sequence.py (hunk 1 above)
Comment (bernt-matthias) [reply lib/galaxy/datatypes/sequence.py]: Sure. I think I just copy pasted this from `FastqSanger`. Fixed it also there.

Comment (jdavcs) [reply lib/galaxy/datatypes/sequence.py]: Thank you! Much more readable now, I think.

**Review by jdavcs — APPROVED** [review]
Looks good to me, +1 (pending tests passing).

**Comment by bgruening** [comment]
Flagging @wm75 here as we had some new super long read data that had some issues.

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 3]
```diff
@@ -124,16 +124,16 @@
...
       <converter file="fastq_to_fqtoc.xml" target_datatype="fqtoc"/>
     </datatype>
-    <datatype extension="fastqcssanger" auto_compressed_types="gz,bz2" type="galaxy.datatypes.sequence:FastqCSSanger" display_in_upload="true">
+    <datatype extension="fastqcssanger" auto_compressed_types="gz,bz2" type="galaxy.datatypes.sequence:FastqCSSanger" display_in_upload="true" description="sequence in in color space phred scored quality values 0:93 represented by ASCII 33:126">
       <converter file="fastq_to_fqtoc.xml" target_datatype="fqtoc"/>
     </datatype>
-    <datatype extension="fastqillumina" auto_compressed_types="gz,bz2" type="galaxy.datatypes.sequence:FastqIllumina" display_in_upload="true">
+    <datatype extension="fastqillumina" auto_compressed_types="gz,bz2" type="galaxy.datatypes.sequence:FastqIllumina" display_in_upload="true" description="Sanger variant of the FASTQ format: phred+33, sequence in color space">
```
Comment (bernt-matthias) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]:
```suggestion
    <datatype extension="fastqillumina" auto_compressed_types="gz,bz2" type="galaxy.datatypes.sequence:FastqIllumina" display_in_upload="true" description="Sanger variant of the FASTQ format: phred+64">
```

**Pull request merged**